pandas/io/common.py

```diff
@@ -158,11 +144,25 @@ def urlopen(*args, **kwargs):
     return urllib.request.urlopen(*args, **kwargs)
 
 
+def is_fsspec_url(url: FilePathOrBuffer) -> bool:
+    """
+    Returns true if fsspec is installed and the given URL looks like
+    something fsspec can handle
+    """
+    try:
+        import fsspec  # noqa: F401
+
+        return isinstance(url, str) and ("::" in url or "://" in url)
```

Comment:
Should this also check if there's an fsspec-compatible implementation for that protocol? What's the behavior / error message if you have `fsspec` installed but not `s3fs` and do `read_csv("s3://...")`?

Reply:
fsspec gives specific error messages for known protocols that are not available due to missing dependency (s3fs would be such a one), and a different message is the protocol is completely unknown.

Reply:
Does that error surface to the user? Or does pandas swallow it somewhere along the way?